relay/polkadot/src/impls.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use super::*;
+use core::marker::PhantomData;
+use frame_support::traits::{tokens::ConversionFromAssetBalance, Contains};
+use primitives::Id as ParaId;
+use xcm_builder::IsChildSystemParachain;
+
+/// Determines if the given `asset_kind` is a native asset. If it is, returns the balance without
+/// conversion; otherwise, delegates to the implementation specified by `I`.
+/// Example where the `asset_kind` represents the native asset:
+/// ```
+/// `asset_kind`: VersionedLocatableAsset: {
+///  `location`: (1, Parachain(1000)), // location of a Sibling Parachain
+///  `asset_id`: (1, Here), // the asset id in the context of `asset_kind.location`
+/// }
+/// ```
+pub struct NativeOnSystemParachain<I>(PhantomData<I>);
+impl<I> ConversionFromAssetBalance<Balance, VersionedLocatableAsset, Balance>
+	for NativeOnSystemParachain<I>
```

Comment:
```suggestion
pub struct NativeOnSystemParachainElseConvertWith<I>(PhantomData<I>);
impl<I> ConversionFromAssetBalance<Balance, VersionedLocatableAsset, Balance>
	for NativeOnSystemParachainElseConvertWith<I>
```

Reply:
I like my version more, I personally prefer short and laconic names with docs, instead long names.
But if you get one more vote for your rename suggestion, I rename

Reply:
> well, I don't mind longer names if they are self-describing :) e.g. NativeOn**Sibling**Parachain does not mention in the docs that it filters just system parachains, so I think at least NativeOn**System**Parachain or NativeOn**SiblingSystem**Parachain is better

@muharem to avoid any confusions (e.g. NativeOnSibling, can be ACA or whatever), why not to just rename all of `NativeOnSiblingParachain` and `NativeOnSystemParachain` to the final `UnityOrOuterConversion` temporary with several impls here? When we do polkadot-sdk bump with https://github.com/paritytech/polkadot-sdk/pull/3659, it wil just "change imports"